hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -180,6 +180,8 @@ public List<ContainerBlockInfo> chooseContainerForBlockDeletion(
       throws StorageContainerException {
     Map<Long, ContainerData> containerDataMap =
         ozoneContainer.getContainerSet().getContainerMap().entrySet().stream()
+            .filter(e -> ((KeyValueContainerData) e.getValue()
+                .getContainerData()).getNumPendingDeletionBlocks() > 0)
```

Comment:
Since we are already filtering ContainerData based on the pending block deletes. We should refactor the code to not check again for `entry.getNumPendingDeletionBlocks() > 0`

https://github.com/apache/ozone/blob/8d331a371fca43f5f7a4cf027f227f237388dc46/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/TopNOrderedContainerDeletionChoosingPolicy.java#L76-L83

Reply:
Done, thanks for the suggestion.

Reply:
NIT: I believe this should not be done it is true currently only BlockDeletingService is calling the function but if there is some other implementation it would ending up doing the calculation anyways. But this should also be fine.

Reply:
> NIT: I believe this should not be done it is true currently only BlockDeletingService is calling the function but if there is some other implementation it would ending up doing the calculation anyways. But this should also be fine

Good point. However, if there is some other implementation, it would probably check some other condition instead of `entry.getNumPendingDeletionBlocks() > 0`.